src/udp_wrap.cc

```diff
@@ -245,27 +246,41 @@ void UDPWrap::DoSend(const FunctionCallbackInfo<Value>& args, int family) {
 
   // send(req, buffer, offset, length, port, address)
   CHECK(args[0]->IsObject());
-  CHECK(Buffer::HasInstance(args[1]));
+  CHECK(args[1]->IsArray());
   CHECK(args[2]->IsUint32());
   CHECK(args[3]->IsUint32());
-  CHECK(args[4]->IsUint32());
-  CHECK(args[5]->IsString());
-  CHECK(args[6]->IsBoolean());
+  CHECK(args[4]->IsString());
+  CHECK(args[5]->IsBoolean());
 
   Local<Object> req_wrap_obj = args[0].As<Object>();
-  Local<Object> buffer_obj = args[1].As<Object>();
-  size_t offset = args[2]->Uint32Value();
-  size_t length = args[3]->Uint32Value();
-  const unsigned short port = args[4]->Uint32Value();
-  node::Utf8Value address(env->isolate(), args[5]);
-  const bool have_callback = args[6]->IsTrue();
-
-  CHECK_LE(length, Buffer::Length(buffer_obj) - offset);
+  Local<Array> chunks = args[1].As<Array>();
+  // it is faster to fetch the length of the
+  // array in js-land
+  size_t count = args[2]->Uint32Value();
+  const unsigned short port = args[3]->Uint32Value();
+  node::Utf8Value address(env->isolate(), args[4]);
+  const bool have_callback = args[5]->IsTrue();
 
   SendWrap* req_wrap = new SendWrap(env, req_wrap_obj, have_callback);
 
-  uv_buf_t buf = uv_buf_init(Buffer::Data(buffer_obj) + offset,
-                             length);
+  // allocate uv_buf_t of the correct size
+  // if bigger than 16 elements
+  uv_buf_t bufs_[16];
+  uv_buf_t* bufs = bufs_;
+
+  if (ARRAY_SIZE(bufs_) < count)
+    bufs = new uv_buf_t[count];
+
+  // construct uv_buf_t array
+  for (size_t i = 0; i < count; i++) {
+    Local<Value> chunk = chunks->Get(i);
+
+    CHECK(Buffer::HasInstance(chunk));
```

Comment:
No need for this. Both `Buffer::Length()` and `Buffer::Data()` already have this check.